bfd/src/main/java/gov/cms/ab2d/bfd/client/BFDSearchImpl.java

```diff
@@ -0,0 +1,75 @@
+package gov.cms.ab2d.bfd.client;
+
+import ca.uhn.fhir.parser.IParser;
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+import java.io.InputStream;
+import java.time.OffsetDateTime;
+
+@Component
+@Slf4j
+public class BFDSearchImpl implements BFDSearch {
+
+    private final HttpClient httpClient;
+
+    private final IParser parser;
+
+    private final Environment environment;
+
+    @Value("${bfd.serverBaseUrl}")
+    private String serverBaseUrl;
+
+    public BFDSearchImpl(HttpClient httpClient, IParser iParser, Environment environment) {
+        this.httpClient = httpClient;
+        this.parser = iParser;
+        this.environment = environment;
+    }
+
+    @Override
+    public Bundle searchEOB(String patientId, OffsetDateTime since, int pageSize) throws IOException {
+        StringBuilder url = new StringBuilder(serverBaseUrl + "ExplanationOfBenefit?patient=" + patientId + "&excludeSAMHSA=true");
+
+        if (since != null) {
+            url.append("&_lastUpdated=ge").append(since);
+        }
+
+        if (pageSize > 0) {
+            url.append("&_count=").append(pageSize);
+        }
+
+        HttpGet request = new HttpGet(url.toString());
```

Comment:
Is there a reason not to use RestTempate?

Reply:
Apache commons is a little easier to use and has better perf. from what I've seen, it's currently what's used in other places